### ISSUE TITLE
关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法

### ISSUE THREAD

**Issue by UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ** [kkevsekk1/Auto.js-VSCode-Extension#29]
需要修改文件 -----\.vscode\extensions\aaroncheng.auto-js-vsce-fixed-1.109.0\out\extension.js
类Extension（98行-135行）中的内容

修改前后对比
![image_2024-03-16_17-43-33](https://github.com/kkevsekk1/Auto.js-VSCode-Extension/assets/101484379/2a8151a3-4a81-4bae-92d1-a49ffbbd5dc5)

**修改完成之后，需要重启vscode。**
```
class Extension {
    constructor() {
        this.documentViewPanel = undefined;
        this.qrCodeViewPanel = undefined;
        this.documentCache = new Map();
    }

    showServerAddress() {
        let servers = server.getIPs().join(":" + server.getPort() + " or ") + ":" + server.getPort();
        vscode.window.showInformationMessage(`Auto.js Autox.js \r\n server running on ${servers}`);
    }

    showQrCode() {
        let ips = server.getIPs();
        if (ips.length == 1) {
            this.showQrcodeWebview(ips[0]);
        } else {
            vscode.window.showQuickPick(ips).then(ip => {
                this.showQrcodeWebview(ip);
            });
        }
    }

    showQrcodeWebview(ip) {
        let url = `ws://${ip}:${server.getPort()}`;
        if (!this.qrCodeViewPanel) {
            this.qrCodeViewPanel = vscode.window.createWebviewPanel('Qr code', "Qr code", vscode.ViewColumn.Beside, {
                enableScripts: true,
            });

            this.qrCodeViewPanel.onDidDispose(() => {
                this.qrCodeViewPanel = undefined;
            }, undefined, exports._context.subscriptions);
        }
        this.qrCodeViewPanel.webview.html = this.getQrCodeHtml(this.qrCodeViewPanel.webview, url);
    }

    getQrCodeHtml(webview, text) {
        const icon = this.getVscodeResourceUrl(webview, "logo.png");
        const qrcodejs = this.getVscodeResourceUrl(webview, "assets/qrcode.js");
        return `<!DOCTYPE html>
```
```
    getVscodeResourceUrl(webview, relativePath) {
        const onDiskPath = vscode.Uri.file(path.join(exports._context.extensionPath, relativePath));
        return webview.asWebviewUri(onDiskPath);
    }
```

**Comment by zlhdd108** [comment]
> 需要修改文件 -----.vscode\extensions\aaroncheng.auto-js-vsce-fixed-1.109.0\out\extension.js 类Extension（98行-135行）中的内容
> 
> ```
> class Extension {
>     constructor() {
>         this.documentViewPanel = undefined;
>         this.qrCodeViewPanel = undefined;
>         this.documentCache = new Map();
>     }
> 
>     showServerAddress() {
>         let servers = server.getIPs().join(":" + server.getPort() + " or ") + ":" + server.getPort();
>         vscode.window.showInformationMessage(`Auto.js Autox.js \r\n server running on ${servers}`);
>     }
> 
>     showQrCode() {
>         let ips = server.getIPs();
>         if (ips.length == 1) {
>             this.showQrcodeWebview(ips[0]);
>         } else {
>             vscode.window.showQuickPick(ips).then(ip => {
>                 this.showQrcodeWebview(ip);
>             });
>         }
>     }
> 
>     showQrcodeWebview(ip) {
>         let url = `ws://${ip}:${server.getPort()}`;
>         if (!this.qrCodeViewPanel) {
>             this.qrCodeViewPanel = vscode.window.createWebviewPanel('Qr code', "Qr code", vscode.ViewColumn.Beside, {
>                 enableScripts: true,
>             });
> 
>             this.qrCodeViewPanel.onDidDispose(() => {
>                 this.qrCodeViewPanel = undefined;
>             }, undefined, exports._context.subscriptions);
>         }
>         this.qrCodeViewPanel.webview.html = this.getQrCodeHtml(this.qrCodeViewPanel.webview, url);
>     }
> 
>     getQrCodeHtml(webview, text) {
>         const icon = this.getVscodeResourceUrl(webview, "logo.png");
>         const qrcodejs = this.getVscodeResourceUrl(webview, "assets/qrcode.js");
>         return `<!DOCTYPE html>
> ```
> 
> ```
>     getVscodeResourceUrl(webview, relativePath) {
>         const onDiskPath = vscode.Uri.file(path.join(exports._context.extensionPath, relativePath));
>         return webview.asWebviewUri(onDiskPath);
>     }
> ```

修改完了，还是不显示二维码……

**Comment by UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ** [comment]
> > 需要修改文件 -----.vscode\extensions\aaroncheng.auto-js-vsce-fixed-1.109.0\out\extension.js 类Extension（98行-135行）中的内容
> > ```
> > class Extension {
> >     constructor() {
> >         this.documentViewPanel = undefined;
> >         this.qrCodeViewPanel = undefined;
> >         this.documentCache = new Map();
> >     }
> > 
> >     showServerAddress() {
> >         let servers = server.getIPs().join(":" + server.getPort() + " or ") + ":" + server.getPort();
> >         vscode.window.showInformationMessage(`Auto.js Autox.js \r\n server running on ${servers}`);
> >     }
> > 
> >     showQrCode() {
> >         let ips = server.getIPs();
> >         if (ips.length == 1) {
> >             this.showQrcodeWebview(ips[0]);
> >         } else {
> >             vscode.window.showQuickPick(ips).then(ip => {
> >                 this.showQrcodeWebview(ip);
> >             });
> >         }
> >     }
> > 
> >     showQrcodeWebview(ip) {
> >         let url = `ws://${ip}:${server.getPort()}`;
> >         if (!this.qrCodeViewPanel) {
> >             this.qrCodeViewPanel = vscode.window.createWebviewPanel('Qr code', "Qr code", vscode.ViewColumn.Beside, {
> >                 enableScripts: true,
> >             });
> > 
> >             this.qrCodeViewPanel.onDidDispose(() => {
> >                 this.qrCodeViewPanel = undefined;
> >             }, undefined, exports._context.subscriptions);
> >         }
> >         this.qrCodeViewPanel.webview.html = this.getQrCodeHtml(this.qrCodeViewPanel.webview, url);
> >     }
> > 
> >     getQrCodeHtml(webview, text) {
> >         const icon = this.getVscodeResourceUrl(webview, "logo.png");
> >         const qrcodejs = this.getVscodeResourceUrl(webview, "assets/qrcode.js");
> >         return `<!DOCTYPE html>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> >     getVscodeResourceUrl(webview, relativePath) {
> >         const onDiskPath = vscode.Uri.file(path.join(exports._context.extensionPath, relativePath));
> >         return webview.asWebviewUri(onDiskPath);
> >     }
> > ```
> 
> 修改完了，还是不显示二维码……

修改完成之后，需要重启vscode。

**Comment by zlhdd108** [comment]
关闭vscode——修改extension.js——进入vscode——无二维码；
重启PC——进入vscode——依然无二维码；
提这个issue之前就是这样操作的




 
发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ
发送时间： 2024-03-16 16:37
收件人： kkevsekk1/Auto.js-VSCode-Extension
抄送： LonelyHao; Comment
主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29)
需要修改文件 -----.vscode\extensions\aaroncheng.auto-js-vsce-fixed-1.109.0\out\extension.js 类Extension（98行-135行）中的内容
class Extension {
    constructor() {
        this.documentViewPanel = undefined;
        this.qrCodeViewPanel = undefined;
        this.documentCache = new Map();
    }
    showServerAddress() {
        let servers = server.getIPs().join(":" + server.getPort() + " or ") + ":" + server.getPort();
        vscode.window.showInformationMessage(`Auto.js Autox.js \r\n server running on ${servers}`);
    }
    showQrCode() {
        let ips = server.getIPs();
        if (ips.length == 1) {
            this.showQrcodeWebview(ips[0]);
        } else {
            vscode.window.showQuickPick(ips).then(ip => {
                this.showQrcodeWebview(ip);
            });
        }
    }
    showQrcodeWebview(ip) {
        let url = `ws://${ip}:${server.getPort()}`;
        if (!this.qrCodeViewPanel) {
            this.qrCodeViewPanel = vscode.window.createWebviewPanel('Qr code', "Qr code", vscode.ViewColumn.Beside, {
                enableScripts: true,
            });
            this.qrCodeViewPanel.onDidDispose(() => {
                this.qrCodeViewPanel = undefined;
            }, undefined, exports._context.subscriptions);
        }
        this.qrCodeViewPanel.webview.html = this.getQrCodeHtml(this.qrCodeViewPanel.webview, url);
    }
    getQrCodeHtml(webview, text) {
        const icon = this.getVscodeResourceUrl(webview, "logo.png");
        const qrcodejs = this.getVscodeResourceUrl(webview, "assets/qrcode.js");
        return `<!DOCTYPE html>
    getVscodeResourceUrl(webview, relativePath) {
        const onDiskPath = vscode.Uri.file(path.join(exports._context.extensionPath, relativePath));
        return webview.asWebviewUri(onDiskPath);
    }

修改完了，还是不显示二维码……
修改完成之后，需要重启vscode。

**Comment by UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ** [comment]
> 关闭vscode——修改extension.js——进入vscode——无二维码； 重启PC——进入vscode——依然无二维码； 提这个issue之前就是这样操作的 发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ 发送时间： 2024-03-16 16:37 收件人： kkevsekk1/Auto.js-VSCode-Extension 抄送： LonelyHao; Comment 主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29) 需要修改文件 -----.vscode\extensions\aaroncheng.auto-js-vsce-fixed-1.109.0\out\extension.js 类Extension（98行-135行）中的内容 class Extension { constructor() { this.documentViewPanel = undefined; this.qrCodeViewPanel = undefined; this.documentCache = new Map(); } showServerAddress() { let servers = server.getIPs().join(":" + server.getPort() + " or ") + ":" + server.getPort(); vscode.window.showInformationMessage(`Auto.js Autox.js \r\n server running on ${servers}`); } showQrCode() { let ips = server.getIPs(); if (ips.length == 1) { this.showQrcodeWebview(ips[0]); } else { vscode.window.showQuickPick(ips).then(ip => { this.showQrcodeWebview(ip); }); } } showQrcodeWebview(ip) { let url = `ws://${ip}:${server.getPort()}`; if (!this.qrCodeViewPanel) { this.qrCodeViewPanel = vscode.window.createWebviewPanel('Qr code', "Qr code", vscode.ViewColumn.Beside, { enableScripts: true, }); this.qrCodeViewPanel.onDidDispose(() => { this.qrCodeViewPanel = undefined; }, undefined, exports._context.subscriptions); } this.qrCodeViewPanel.webview.html = this.getQrCodeHtml(this.qrCodeViewPanel.webview, url); } getQrCodeHtml(webview, text) { const icon = this.getVscodeResourceUrl(webview, "logo.png"); const qrcodejs = this.getVscodeResourceUrl(webview, "assets/qrcode.js"); return `<!DOCTYPE html> getVscodeResourceUrl(webview, relativePath) { const onDiskPath = vscode.Uri.file(path.join(exports._context.extensionPath, relativePath)); return webview.asWebviewUri(onDiskPath); } 修改完了，还是不显示二维码…… 修改完成之后，需要重启vscode。 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>

![image_2024-03-16_17-43-33](https://github.com/kkevsekk1/Auto.js-VSCode-Extension/assets/101484379/2a8151a3-4a81-4bae-92d1-a49ffbbd5dc5)
这是修改前后的对比，可以看一下有没有搞错了，如果还不行我就不清楚了。

这是我用的vscode关于页的信息
版本: 1.87.2 (user setup)
提交: 863d2581ecda6849923a2118d93a088
b0745d9d6
日期: 2024-03-08T15:20:17.278Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.22631

**Comment by zlhdd108** [comment]
是的，我也做过这个对比；而且有意思的是，自己的电脑上面，不修改也能正常显示二维码；办公电脑上就有这个问题




 
发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ
发送时间： 2024-03-16 17:46
收件人： kkevsekk1/Auto.js-VSCode-Extension
抄送： LonelyHao; Comment
主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29)
关闭vscode——修改extension.js——进入vscode——无二维码； 重启PC——进入vscode——依然无二维码； 提这个issue之前就是这样操作的 发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ 发送时间： 2024-03-16 16:37 收件人： kkevsekk1/Auto.js-VSCode-Extension 抄送： LonelyHao; Comment 主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29) 需要修改文件 -----.vscode\extensions\aaroncheng.auto-js-vsce-fixed-1.109.0\out\extension.js 类Extension（98行-135行）中的内容 class Extension { constructor() { this.documentViewPanel = undefined; this.qrCodeViewPanel = undefined; this.documentCache = new Map(); } showServerAddress() { let servers = server.getIPs().join(":" + server.getPort() + " or ") + ":" + server.getPort(); vscode.window.showInformationMessage(Auto.js Autox.js \r\n server running on ${servers}); } showQrCode() { let ips = server.getIPs(); if (ips.length == 1) { this.showQrcodeWebview(ips[0]); } else { vscode.window.showQuickPick(ips).then(ip => { this.showQrcodeWebview(ip); }); } } showQrcodeWebview(ip) { let url = ws://${ip}:${server.getPort()}; if (!this.qrCodeViewPanel) { this.qrCodeViewPanel = vscode.window.createWebviewPanel('Qr code', "Qr code", vscode.ViewColumn.Beside, { enableScripts: true, }); this.qrCodeViewPanel.onDidDispose(() => { this.qrCodeViewPanel = undefined; }, undefined, exports._context.subscriptions); } this.qrCodeViewPanel.webview.html = this.getQrCodeHtml(this.qrCodeViewPanel.webview, url); } getQrCodeHtml(webview, text) { const icon = this.getVscodeResourceUrl(webview, "logo.png"); const qrcodejs = this.getVscodeResourceUrl(webview, "assets/qrcode.js"); return ` getVscodeResourceUrl(webview, relativePath) { const onDiskPath = vscode.Uri.file(path.join(exports._context.extensionPath, relativePath)); return webview.asWebviewUri(onDiskPath); } 修改完了，还是不显示二维码…… 修改完成之后，需要重启vscode。 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***>
image_2024-03-16_17-43-33.png (view on web)
这是修改前后的对比，可以看一下有没有搞错了，如果还不行我就不清楚了。
这是我用的vscode关于页的信息
版本: 1.87.2 (user setup)
提交: 863d2581ecda6849923a2118d93a088
b0745d9d6
日期: 2024-03-08T15:20:17.278Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.22631

**Comment by UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ** [comment]
> 是的，我也做过这个对比；而且有意思的是，自己的电脑上面，不修改也能正常显示二维码；办公电脑上就有这个问题 发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ 发送时间： 2024-03-16 17:46 收件人： kkevsekk1/Auto.js-VSCode-Extension 抄送： LonelyHao; Comment 主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29) 关闭vscode——修改extension.js——进入vscode——无二维码； 重启PC——进入vscode——依然无二维码； 提这个issue之前就是这样操作的 发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ 发送时间： 2024-03-16 16:37 收件人： kkevsekk1/Auto.js-VSCode-Extension 抄送： LonelyHao; Comment 主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29) 需要修改文件 -----.vscode\extensions\aaroncheng.auto-js-vsce-fixed-1.109.0\out\extension.js 类Extension（98行-135行）中的内容 class Extension { constructor() { this.documentViewPanel = undefined; this.qrCodeViewPanel = undefined; this.documentCache = new Map(); } showServerAddress() { let servers = server.getIPs().join(":" + server.getPort() + " or ") + ":" + server.getPort(); vscode.window.showInformationMessage(Auto.js Autox.js \r\n server running on ${servers}); } showQrCode() { let ips = server.getIPs(); if (ips.length == 1) { this.showQrcodeWebview(ips[0]); } else { vscode.window.showQuickPick(ips).then(ip => { this.showQrcodeWebview(ip); }); } } showQrcodeWebview(ip) { let url = ws://${ip}:${server.getPort()}; if (!this.qrCodeViewPanel) { this.qrCodeViewPanel = vscode.window.createWebviewPanel('Qr code', "Qr code", vscode.ViewColumn.Beside, { enableScripts: true, }); this.qrCodeViewPanel.onDidDispose(() => { this.qrCodeViewPanel = undefined; }, undefined, exports._context.subscriptions); } this.qrCodeViewPanel.webview.html = this.getQrCodeHtml(this.qrCodeViewPanel.webview, url); } getQrCodeHtml(webview, text) { const icon = this.getVscodeResourceUrl(webview, "logo.png"); const qrcodejs = this.getVscodeResourceUrl(webview, "assets/qrcode.js"); return ` getVscodeResourceUrl(webview, relativePath) { const onDiskPath = vscode.Uri.file(path.join(exports._context.extensionPath, relativePath)); return webview.asWebviewUri(onDiskPath); } 修改完了，还是不显示二维码…… 修改完成之后，需要重启vscode。 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***> image_2024-03-16_17-43-33.png (view on web) 这是修改前后的对比，可以看一下有没有搞错了，如果还不行我就不清楚了。 这是我用的vscode关于页的信息 版本: 1.87.2 (user setup) 提交: 863d2581ecda6849923a2118d93a088 b0745d9d6 日期: 2024-03-08T15:20:17.278Z Electron: 27.3.2 ElectronBuildId: 26836302 Chromium: 118.0.5993.159 Node.js: 18.17.1 V8: 11.8.172.18-electron.0 OS: Windows_NT x64 10.0.22631 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>


你的电脑不修改也能用 可能是因为Auto.js-Autox.js-VSCodeExt更新到了v1.110.0

**Comment by zlhdd108** [comment]
诶，确实是的；怪了，自己的电脑还是上周用的；工作机是昨天用的，还点了个更新，都没有升级……




 
发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ
发送时间： 2024-03-16 17:59
收件人： kkevsekk1/Auto.js-VSCode-Extension
抄送： LonelyHao; Comment
主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29)
是的，我也做过这个对比；而且有意思的是，自己的电脑上面，不修改也能正常显示二维码；办公电脑上就有这个问题 发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ 发送时间： 2024-03-16 17:46 收件人： kkevsekk1/Auto.js-VSCode-Extension 抄送： LonelyHao; Comment 主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29) 关闭vscode――修改extension.js――进入vscode――无二维码； 重启PC――进入vscode――依然无二维码； 提这个issue之前就是这样操作的 发件人： UGYnpU2nuB8QCFqDrsWZgmxMBp43WQ 发送时间： 2024-03-16 16:37 收件人： kkevsekk1/Auto.js-VSCode-Extension 抄送： LonelyHao; Comment 主题： Re: [kkevsekk1/Auto.js-VSCode-Extension] 关于Auto.js-Autox.js-VSCodeExt v1.109.0不显示二维码的临时解决办法 (Issue #29) 需要修改文件 -----.vscode\extensions\aaroncheng.auto-js-vsce-fixed-1.109.0\out\extension.js 类Extension（98行-135行）中的内容 class Extension { constructor() { this.documentViewPanel = undefined; this.qrCodeViewPanel = undefined; this.documentCache = new Map(); } showServerAddress() { let servers = server.getIPs().join(":" + server.getPort() + " or ") + ":" + server.getPort(); vscode.window.showInformationMessage(Auto.js Autox.js \r\n server running on ${servers}); } showQrCode() { let ips = server.getIPs(); if (ips.length == 1) { this.showQrcodeWebview(ips[0]); } else { vscode.window.showQuickPick(ips).then(ip => { this.showQrcodeWebview(ip); }); } } showQrcodeWebview(ip) { let url = ws://${ip}:${server.getPort()}; if (!this.qrCodeViewPanel) { this.qrCodeViewPanel = vscode.window.createWebviewPanel('Qr code', "Qr code", vscode.ViewColumn.Beside, { enableScripts: true, }); this.qrCodeViewPanel.onDidDispose(() => { this.qrCodeViewPanel = undefined; }, undefined, exports._context.subscriptions); } this.qrCodeViewPanel.webview.html = this.getQrCodeHtml(this.qrCodeViewPanel.webview, url); } getQrCodeHtml(webview, text) { const icon = this.getVscodeResourceUrl(webview, "logo.png"); const qrcodejs = this.getVscodeResourceUrl(webview, "assets/qrcode.js"); return ` getVscodeResourceUrl(webview, relativePath) { const onDiskPath = vscode.Uri.file(path.join(exports._context.extensionPath, relativePath)); return webview.asWebviewUri(onDiskPath); } 修改完了，还是不显示二维码…… 修改完成之后，需要重启vscode。 ― Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.> image_2024-03-16_17-43-33.png (view on web) 这是修改前后的对比，可以看一下有没有搞错了，如果还不行我就不清楚了。 这是我用的vscode关于页的信息 版本: 1.87.2 (user setup) 提交: 863d2581ecda6849923a2118d93a088 b0745d9d6 日期: 2024-03-08T15:20:17.278Z Electron: 27.3.2 ElectronBuildId: 26836302 Chromium: 118.0.5993.159 Node.js: 18.17.1 V8: 11.8.172.18-electron.0 OS: Windows_NT x64 10.0.22631 ― Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.>
你的电脑不修改也能用 可能是因为Auto.js-Autox.js-VSCodeExt更新到了v1.110.0
―
